ethmonitor/chain.go

```diff
@@ -55,6 +59,15 @@ func (c *Chain) push(nextBlock *Block) error {
 		c.blocks = c.blocks[1:]
 	}
 
+	// Update average block time
+	if n > 0 {
+		if c.averageBlockTime == 0 {
+			c.averageBlockTime = float64(nextBlock.Time() - headBlock.Time())
+		} else {
+			c.averageBlockTime = (c.averageBlockTime + float64(nextBlock.Time()-headBlock.Time())) / 2
+		}
```

Comment:
I think you could just move this into the `if` statement above instead.

Reply:
Also, what happens if `c.blocks` is a sequence of block removals due to reorg? I think it might be an issue when updating the average block time.

Reply:
Something else to think about is maybe we should consider computing average block time alternatively as (head block time - starting block time) / (head block number - starting block number) rather than doing a running average.

Reply:
> Also, what happens if `c.blocks` is a sequence of block removals due to reorg? I think it might be an issue when updating the average block time.

Actually, I suppose if there's a reorg, there should still be more added blocks than removed blocks, so the last block in the chain should still be an addition. So this _should_ be OK.

Reply:
> Something else to think about is maybe we should consider computing average block time alternatively as (next block time - starting block time) / (next block number - starting block number) rather than doing a running average.

I think the running average is fine too actually. It means we end up giving more recent measurements higher weight.

Reply:
We can think about the running average, as it will be more accurate for our needs. 
Maybe we can have a flag that we can pass, if we need a running average, or calculate the average from only the last `n` blocks. 🤔

Reply:
I think your current implementation is OK here as-is. We can refine later if we see unusual behaviour.